crates/router/src/core/payments/helpers.rs

```diff
@@ -1227,72 +1246,122 @@ pub async fn make_pm_data<'a, F: Clone, R>(
 
     // TODO: Handle case where payment method and token both are present in request properly.
     let payment_method = match (request, hyperswitch_token) {
-        (_, Some(hyperswitch_token)) => {
-            let (pm, supplementary_data) = vault::Vault::get_payment_method_data_from_locker(
-                state,
-                &hyperswitch_token,
-            )
-            .await
-            .attach_printable(
-                "Payment method for given token not found or there was a problem fetching it",
-            )?;
+        (_, Some(hyperswitch_token)) => match hyperswitch_token.kind {
```

Comment:
Avoid writing long implementation, split branching into separate functions and writing comments for those.